source/read/change-streams.txt

```diff
@@ -0,0 +1,261 @@
+         this option, see :ref:`php-change-stream-pre-post-image`.
+    
+   * - ``resumeAfter``
+     - | Instructs ``watch()`` to resume returning changes after the
+         operation specified in the resume token.
+       | Each change stream event document includes a resume token as the ``_id``
+         field. Pass the entire ``_id`` field of the change event document that
+         represents the operation you want to resume after.
+       | This option is mutually exclusive with ``startAfter`` and ``startAtOperationTime``.
+
+   * - ``startAfter``
+     - | Instructs ``watch()`` to start a new change stream after the
+         operation specified in the resume token. This field allows notifications to
+         resume after an invalidate event.
+       | Each change stream event document includes a resume token as the ``_id``
+         field. Pass the entire ``_id`` field of the change event document that
+         represents the operation you want to resume after.
+       | This option is mutually exclusive with ``resumeAfter`` and ``startAtOperationTime``.
+
+   * - ``startAtOperationTime``
+     - | Instructs the change stream to only provide changes that occurred at or after
+         the specified timestamp. 
+       | This option is mutually exclusive with ``startAfter`` and ``resumeAfter``.
+
+   * - ``typeMap``
+     - | The type map to apply to cursors, which determines how BSON documents are converted
+         to PHP values.
+
+   * - ``collation``
+     - | Sets the collation to use for the change stream cursor.
+
+For a full list of ``watch()`` options, see `MongoDB\\Collection::watch()
+<{+api+}/method/MongoDBCollection-watch/>`__ in the API
+documentation.
+
+.. _php-change-stream-pre-post-image:
+
+Include Pre-Images and Post-Images
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. important::
+
+   You can enable pre-images and post-images on collections only if your
+   deployment uses MongoDB v6.0 or later.
+
+By default, when you perform an operation on a collection, the
+corresponding change event includes only the delta of the fields
+modified by that operation. To see the full document before or after a
+change, specify the ``fullDocumentBeforeChange`` or the ``fullDocument``
+options in an array parameter to ``watch()``.
+
+The **pre-image** is the full version of a document *before* a change. To include the
+pre-image in the change stream event, set the ``fullDocumentBeforeChange`` option
+to one of the following values:
+
+- ``MongoDB\Operation\Watch::FULL_DOCUMENT_BEFORE_CHANGE_WHEN_AVAILABLE``: The change event includes
+  a pre-image of the modified document for change events. If the pre-image is not available, this
+  change event field has a ``null`` value.
+- ``MongoDB\Operation\Watch::FULL_DOCUMENT_BEFORE_CHANGE_REQUIRED``: The change event includes a pre-image
+  of the modified document for change events. If the pre-image is not available, the
+  {+php-library+} raises an error.  
```

Comment:
Technically, the server raises errors for this and the corresponding post-image case. PHPLIB would just propagate that error to the user. Not sure if this is important to clarify, as this might mislead users into thinking it's a client-side error.

Reply:
Did you mean to change the text here to clarify that PHPLIB will propagate the server-side error?

Reply:
oops yes I did, fixed now